DQM/CastorMonitor/src/CastorDigiMonitor.cc

```diff
@@ -81,38 +122,53 @@ void CastorDigiMonitor::bookHistograms(DQMStore::IBooker& ibooker,
 	hSector->getTH1F()->GetYaxis()->SetTitle("QIE(fC)");
 
-    sprintf(s,"QfC=f(x=Tile y=TS) (cumulative)");
-      h2QtsvsCh = ibooker.book2D(s,s,224,0.,224., 10,0.,10.);
-      h2QtsvsCh->getTH2F()->GetXaxis()->SetTitle(sTileIndex);
-      h2QtsvsCh->getTH2F()->GetYaxis()->SetTitle("TS");
-      h2QtsvsCh->getTH2F()->SetOption("colz");
-
-    sprintf(s,"QmeanfC=f(Tile TS)");
-      h2QmeantsvsCh = ibooker.book2D(s,s,224,0.,224., 10,0.,10.);
-      h2QmeantsvsCh->getTH2F()->GetXaxis()->SetTitle(sTileIndex);
-      h2QmeantsvsCh->getTH2F()->GetYaxis()->SetTitle("Time Slice");
-      h2QmeantsvsCh->getTH2F()->SetOption("colz");
-
-    sprintf(s,"QrmsfC=f(Tile TS)");
-      h2QrmsTSvsCh = ibooker.book2D(s,s,224,0.,224., 10,0.,10.);
-      h2QrmsTSvsCh->getTH2F()->GetXaxis()->SetTitle(sTileIndex);
-      h2QrmsTSvsCh->getTH2F()->GetYaxis()->SetTitle("TS");
+    st = "Castor cells avr digi(fC) per event Map TS vs Channel";
+    h2QmeantsvsCh = ibooker.bookProfile2D(st,st+";"+string(sTileIndex)+";TS",
+        224,0.,224., 10,0.,10., 0.,1.e10,"");
+      h2QmeantsvsCh->getTProfile2D()->SetOption("colz");
+
+    st = "Castor cells avr digiRMS(fC) per event Map TS vs Channel";
+      h2QrmsTSvsCh = ibooker.book2D(st,st+";"+string(sTileIndex)+";TS",
+	224,0.,224., 10,0.,10.);
       h2QrmsTSvsCh->getTH2F()->SetOption("colz");
 
- sprintf(s,"CASTOR data quality");
+   sprintf(s,"CASTOR data quality");
     h2qualityMap = ibooker.book2D(s,s,14, 0,14, 16, 0,16);
     h2qualityMap->getTH2F()->GetXaxis()->SetTitle("module Z");
     h2qualityMap->getTH2F()->GetYaxis()->SetTitle("Sector #phi");
     h2qualityMap->getTH2F()->SetOption("colz");
     
    hReport = ibooker.bookFloat("CASTOR reportSummary");
 
- sprintf(s,"QmeanfC_map(allTS)");
+   sprintf(s,"QmeanfC_map(allTS)");
       h2QmeanMap = ibooker.book2D(s,s,14,0.,14., 16,0.,16.);
       h2QmeanMap->getTH2F()->GetXaxis()->SetTitle("Module Z");
       h2QmeanMap->getTH2F()->GetYaxis()->SetTitle("Sector #phi");
       h2QmeanMap->getTH2F()->SetOption("textcolz");
 
+ const int NEtow = 20;
+  float EhadTow[NEtow+1];
+  float EMTow[NEtow+1];
+  float ETower[NEtow+2];
+  double E0tow = 500./1024.;
+  EMTow[0] = 0.; EMTow[1] = E0tow;
+  EhadTow[0] = 0.; EhadTow[1] = E0tow;
+  ETower[0] = 0.; ETower[1] = E0tow;
+  double lnBtow = log(1.8); //2.
+  for(int j=1; j<NEtow; j++) EMTow[j+1] = E0tow*exp(j*lnBtow);
+  for(int j=1; j<NEtow; j++) EhadTow[j+1] = E0tow*exp(j*lnBtow);
+  for(int j=1; j<=NEtow; j++) ETower[j+1] = E0tow*exp(j*lnBtow);
+
+  sprintf(s,"CASTOR_Tower_EMvsEhad(fC)");
+    h2towEMvsHAD = ibooker.book2D(s,s, NEtow, EhadTow, NEtow, EMTow);
+    h2towEMvsHAD->getTH2F()->GetXaxis()->SetTitle("Ehad [fC]");
+    h2towEMvsHAD->getTH2F()->GetYaxis()->SetTitle("EM [fC]");
+    h2towEMvsHAD->getTH2F()->SetOption("colz");
+
+ sprintf(s,"CASTOR_TowerTotalEnergy(fC)");
+   htowE = ibooker.book1D(s,s,NEtow+1,ETower);
+   htowE->getTH1F()->GetXaxis()->SetTitle("fC");
+
```

Comment:
Please take a look at the indentation over here. Lines bellow `sprintf`s should be on the same level as `sprintf`s themselves.